src/pykoala/plotting/plot_plot.py

```diff
@@ -0,0 +1,418 @@
+       if type(fig_size) == list:
+           plt.figure(figsize=(fig_size[0], fig_size[1]))
+       else:     
+           plt.figure(figsize=(fig_size, fig_size/2.5))
+   
+   if np.isscalar(x[0]) :
+       xx=[]
+       for i in range(len(y)):
+           xx.append(x)
+   else:
+       xx=x
+
+   if xmin is None : xmin = np.nanmin(xx[0])
+   if xmax is None : xmax = np.nanmax(xx[0])  
+   
+   alpha_=alpha
+   psym_=psym
+   label_=label
+   linewidth_=linewidth
+   markersize_=markersize
+   linestyle_=linestyle
+
+   n_plots=len(y)
+       
+   if np.isscalar(y[0]) ==  False:
+       if np.isscalar(alpha):        
+           if alpha_ == "":
+               alpha =[0.5]*n_plots
+           else:
+               alpha=[alpha_]*n_plots
+       if np.isscalar(psym): psym=[psym_]*n_plots
+       if np.isscalar(label): label=[label_]*n_plots
+       if np.isscalar(linewidth): linewidth=[linewidth_]*n_plots
+       if np.isscalar(markersize):markersize=[markersize_]*n_plots
+       if np.isscalar(linestyle): linestyle=[linestyle_]*n_plots
+       if color == "blue" : 
+           color_list = ["red","blue","green","k","orange", "purple", "cyan", "lime"]
+           color = []
+           j=0
+           for i in range(n_plots):
+               color.append(color_list[j])
+               j = j+1
+               if j == 8 : j = 0
+           
+           #color = [color_list[i] for i in range(n_plots)]
+       if ymax is None: y_max_list = []
+       if ymin is None: y_min_list = []
+              
+       if fcal is None:
+           if np.nanmedian(np.abs(y[0])) < 1E-10:
+               fcal = True
+               if np.nanmedian(y[0]) < 1E-20 and np.var(y[0]) > 0 : fcal = False
+       for i in range(len(y)):
+           if psym[i] == "":
+               plt.plot(xx[i],y[i], color=color[i], alpha=alpha[i], label=label[i], linewidth=linewidth[i], linestyle=linestyle[i])
+           else:
+               #print(xx)
+               #print(y)
+               # print(psym)
+               # print(color) 
+               # print(alpha)
+               # print(label)
+               # print(linewidth)
+               # print(markersize)
+               # print(len(xx),len(y), len(psym), len(color), len(alpha), len(label), len(linewidth), len(markersize))
+               plt.plot(xx[i],y[i], psym[i], color=color[i], alpha=alpha[i], label=label[i], mew=linewidth[i], markersize=markersize[i])
+
+           if ymax == "":
+                    y_max_ = []                
+                    y_max_.extend((y[i][j]) for j in range(len(xx[i])) if (xx[i][j] > xmin and xx[i][j] < xmax) )  
+                    y_max_list.append(np.nanpercentile(y_max_, percentile_max))
+           if ymin == "":
+                    y_min_ = []                
+                    y_min_.extend((y[i][j]) for j in range(len(xx[i])) if (xx[i][j] > xmin and xx[i][j] < xmax) )  
+                    y_min_list.append(np.nanpercentile(y_min_, percentile_min))
```

Comment:
Do you mean `ymin, ymax = np.nanpercentile(spectra_list, percentile_min, percentile_max)`?